pkgs/build-support/rust/lib/default.nix

```diff
@@ -124,23 +124,7 @@ rec {
     #
     setEnv = ''
     env \
-    ''
-    # Due to a bug in how splicing and targetPackages works, in
-    # situations where targetPackages is irrelevant
-    # targetPackages.stdenv.cc is often simply wrong.  We must omit
-    # the following lines when rustTargetPlatform collides with
-    # rustHostPlatform.
-    + lib.optionalString (rustTargetPlatform != rustHostPlatform) ''
-      "CC_${toRustTargetForUseInEnvVars stdenv.targetPlatform}=${ccForTarget}" \
-      "CXX_${toRustTargetForUseInEnvVars stdenv.targetPlatform}=${cxxForTarget}" \
-      "CARGO_TARGET_${toRustTargetForUseInEnvVars stdenv.targetPlatform}_LINKER=${ccForTarget}" \
```

Comment:
I'm really sure there was some obscure case that needed this (the `CARGO_TARGET_${targetPlatform}_LINKER`) variable, but it was super-weird.  I think it had to do with setup hooks using python that used maturin that used Rust.

I can try to find the exact case that needs it.